drivers/ethernet/eth_adin2111.c

```diff
@@ -968,21 +1137,49 @@ static const struct ethernet_api adin2111_port_api = {
 				 NET_L2_GET_CTX_TYPE(ETHERNET_L2), NET_ETH_MTU);
 
 #define ADIN2111_SPI_OPERATION ((uint16_t)(SPI_OP_MODE_MASTER | SPI_TRANSFER_MSB | SPI_WORD_SET(8)))
-
 #define ADIN2111_MAC_INITIALIZE(inst, dev_id, ifaces, name)					\
 	static uint8_t __aligned(4) name##_buffer_##inst[CONFIG_ETH_ADIN2111_BUFFER_SIZE];	\
 	static const struct adin2111_config name##_config_##inst = {				\
-		.id = dev_id,								\
-		.spi = SPI_DT_SPEC_INST_GET(inst, ADIN2111_SPI_OPERATION, 1),			\
+		.id = dev_id,									\
+		.spi = {									\
+			.bus = DEVICE_DT_GET(DT_INST_BUS(inst)),				\
+			.config = {								\
+				.frequency = DT_INST_PROP(inst, spi_max_frequency),		\
+				.operation = (ADIN2111_SPI_OPERATION) |				\
+					DT_INST_PROP(inst, duplex) |				\
+					DT_INST_PROP(inst, frame_format) |			\
+					COND_CODE_1(DT_INST_PROP(inst, spi_cpol),		\
+						    SPI_MODE_CPOL, (0)) |			\
+					COND_CODE_1(DT_INST_PROP(inst, spi_cpha),		\
+						    SPI_MODE_CPHA, (0)) |			\
+					COND_CODE_1(DT_INST_PROP(inst, spi_hold_cs),		\
+						    SPI_HOLD_ON_CS, (0)),			\
+				.slave = DT_REG_ADDR(DT_DRV_INST(inst)),			\
+				.cs = {								\
+					.gpio = COND_CODE_1(DT_INST_PROP_OR(inst, spi_oa, 0),	\
+						({0, 0, 0}),					\
+						(SPI_CS_GPIOS_DT_SPEC_GET(DT_DRV_INST(inst)))),	\
```

Comment:
if the whole objective of `spi-oa` is to override CS control, why not enforce it via a build assert, eg

```
BUILD_ASSERT(DT_INST_HAS_PROP(inst, spi_oa) && !DT_PROP_HAS_IDX(DT_INST_BUS(inst), DT_INST_REG_ADDR(inst));
```
then simply initialize SPI with `SPI_DT_SPEC_INST_GET`.

Another option would be to modify SPI API with a flag to override CS control even if non-NULL.

Reply:
ping here